adi/adpd1080.py

```diff
@@ -0,0 +1,110 @@
+# PARTICULAR PURPOSE ARE DISCLAIMED.
+#
+# IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, INTELLECTUAL PROPERTY
+# RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR
+# BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+# THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from adi.attribute import attribute
+from adi.context_manager import context_manager
+from adi.rx_tx import rx
+
+
+class adpd1080(rx, context_manager):
+
+    """ADPD1080 photo-electronic device."""
+
+    channel = []  # type: ignore
+    _device_name = ""
+    _rx_data_type = ["<u4", "<u4", "<u4", "<u4", "<u4", "<u4", "<u4", "<u4"]
+
+    def __init__(self, uri="", device_index=0):
+        """ADPD1080 class constructor."""
+        context_manager.__init__(self, uri, self._device_name)
+
+        compatible_parts = ["adpd1080"]
+
+        self._ctrl = None
+        index = 0
+
+        # Selecting the device_index-th device from the adpd188 family as working device.
+        for device in self._ctx.devices:
+            if device.name in compatible_parts:
+                if index == device_index:
+                    self._ctrl = device
+                    self._rxadc = device
+                    break
+                else:
+                    index += 1
+
+        # dynamically get channels and sorting them after the color
+        # self._ctrl.channels.sort(key=lambda x: str(x.id[14:]))
+
+        for ch in self._ctrl._channels:
+            name = ch._id
+            self._rx_channel_names.append(name)
+            self.channel.append(self._channel(self._ctrl, name))
+        rx.__init__(self)
+        self.rx_buffer_size = 16
+
+    def rx(self):
+            buff = super().rx()
+            del self._rx__rxbuf
+            return buff
+
+    class _channel(attribute):
+
+        """ADPD1080 channel."""
+
+        def __init__(self, ctrl, channel_name):
+            self.name = channel_name
+            self._ctrl = ctrl
+
+        @property
+        def raw(self):
+            """ADPD1080 channel raw value."""
+            return self._get_iio_attr(self.name, "raw", False)
+
+        @property
+        def offset(self):
+            """ADPD1080 channel offset."""
+            return self._get_iio_attr(self.name, "offset", False)
+
+        @offset.setter
+        def offset(self, value):
+            self._set_iio_attr(self.name, "offset", False, value)
+
+        @property
+        def sample_rate(self):
+            """Sets sampling frequency of the ADPD1080."""
+            return self._get_iio_attr(self.name, "sampling_frequency", False)
+
+        @sample_rate.setter
+        def sample_rate(self, value):
```

Comment:
Is sample rate the same across all channels? If so can we move this up out of the individual channels and just have 1?

Reply:
Will do.